plugins/async-node/ios/Sources/AsyncNodePlugin.swift

```diff
@@ -42,58 +43,70 @@ public class AsyncNodePlugin: JSBasePlugin, NativePlugin {
-                    let encoder = JSONEncoder()
-                    do {
-                        let res = try encoder.encode(encodable)
-                        return context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
-                    } catch {
-                        break
-                    }
-                case .concrete(let jsValue):
-                    return jsValue
-                }
-            }
-
-            return nil
-        })
+           hooks?.onAsyncNode.tap({ node in
+               // hook value is the original node
+               guard let asyncHookHandler = self.asyncHookHandler else {
+                   return JSValue()
+               }
+
+               let replacementNode = try await (asyncHookHandler)(node)
+               var result: JSValue?
+
+               switch replacementNode {
+               case .multiNode(let replacementNodes):
+                   let jsValueArray = replacementNodes.compactMap({ node in
+                       switch node {
+                       case .concrete(let jsValue):
+                           return jsValue
+                       case .encodable(let encodable):
+                           let encoder = JSONEncoder()
+                           do {
+                               let res = try encoder.encode(encodable)
+                               result = context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
+                               return result
+                           } catch {
+                               return nil
+                           }
+                       }
+                   })
+
+                   result = context.objectForKeyedSubscript("Array").objectForKeyedSubscript("from").call(withArguments: [jsValueArray])
+
+               case .singleNode(let replacementNode):
+                   switch replacementNode {
+                   case .encodable(let encodable):
+                       let encoder = JSONEncoder()
+                       do {
+                           let res = try encoder.encode(encodable)
+                           result = context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
+                       } catch {
+                           result = nil
+                       }
+                   case .concrete(let jsValue):
+                       return jsValue
+                   }
+
+              case .nullOrUndefinedNode(let replacementNode):
+                  switch replacementNode {
+                  case .encodable(let encodable):
+                      let encoder = JSONEncoder()
+                          let res = try encoder.encode(encodable)
+                          result = context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
+                  case .concrete(let jsValue):
+                      return jsValue
+                  }
```

Comment:
Totally agree that its actually getting handled by other cases , so I was thinking across that do we actually need to handle this case explicitly since its already being handled !